sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -389,6 +389,23 @@ case class DataSource(
   }
 
   /**
+   * Return all paths represented by the wildcard string.
+   */
+  private def getGlobbedPaths(qualified: Path): Seq[Path] = {
```

Comment:
You are right.
I'll fix this and also limit the max parallelism num in next patch, reuse the config in `InMemoryFileIndex.bulkListLeafFiles`.